[receiver/googlecloudspanner] Test TestItemCardinalityFilter_Filter fails intermittently on Windows runs

### Component(s)

receiver/googlecloudspanner

### Describe the issue you're reporting

[CI/CD failure link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694703823/job/23844185019?pr=32176)

Failure output:
```
Running target 'test' in module 'receiver/googlecloudspannerreceiver' as part of group 'receiver-1'
C:/mingw64/bin/make --no-print-directory -C receiver/googlecloudspannerreceiver test
D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout [30](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694703823/job/23844185019?pr=32176#step:7:31)0s -parallel 4 --tags=""
✓  internal/filterfactory (cached)
✓  internal/metadata (cached)
✖  internal/filter (434ms)
✓  internal/datasource (2.156s)
✓  . (5.051s)
✓  internal/metadataconfig (1.778s)
✓  internal/metadataparser (1.744s)
✓  internal/statsreader (1.891s)

DONE 288 tests, 1 skipped, 1 failure in 13.253s

✖  internal/filter (888ms)

=== Skipped
=== SKIP: internal/statsreader TestStatsReaders_Read (0.00s)
    statsreaders_mockedspanner_test.go:123: Flaky test - See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6[31](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694703823/job/23844185019?pr=32176#step:7:32)8

=== Failed
=== FAIL: internal/filter TestItemCardinalityFilter_Filter (0.10s)
    logger.go:146: 2024-04-15T19:22:05.611Z	DEBUG	Added item to cache	{"seriesKey": "key1", "timestamp": "2021-10-13T20:30:00.000Z"}
    logger.go:146: 2024-04-15T19:22:05.660Z	DEBUG	Added item to cache	{"seriesKey": "key2", "timestamp": "2021-10-13T20:30:00.000Z"}
    logger.go:146: 2024-04-15T19:22:05.661Z	DEBUG	Skip item	{"seriesKey": "key3", "timestamp": "2021-10-13T20:30:00.000Z"}
    logger.go:146: 2024-04-15T19:22:05.661Z	DEBUG	Added item to cache	{"seriesKey": "key5", "timestamp": "2021-10-13T20:30:05.000Z"}
    logger.go:146: 2024-04-15T19:22:05.661Z	DEBUG	Added item to cache	{"seriesKey": "key4", "timestamp": "2021-10-13T20:30:10.000Z"}
    logger.go:146: 2024-04-15T19:22:05.661Z	DEBUG	Skip item	{"seriesKey": "key6", "timestamp": "2021-10-13T20:30:10.000Z"}
    logger.go:146: 2024-04-15T19:22:05.662Z	DEBUG	Added item to cache	{"seriesKey": "key3", "timestamp": "2021-10-13T20:30:20.000Z"}
    logger.go:146: 2024-04-15T19:22:05.662Z	DEBUG	Skip item	{"seriesKey": "key6", "timestamp": "2021-10-13T20:30:20.000Z"}
    itemcardinality_test.go:1[43](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694703823/job/23844185019?pr=32176#step:7:44): 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go:143
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestItemCardinalityFilter_Filter
    logger.go:1[46](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8694703823/job/23844185019?pr=32176#step:7:47): 2024-04-15T19:22:05.716Z	DEBUG	Added item to cache	{"seriesKey": "key3", "timestamp": "2021-10-13T20:30:20.000Z"}
    logger.go:146: 2024-04-15T19:22:05.716Z	DEBUG	Added item to cache	{"seriesKey": "key6", "timestamp": "2021-10-13T20:30:20.000Z"}

DONE 2 runs, 289 tests, 1 skipped, 1 failure in 15.777s
make[2]: *** [../../Makefile.Common:126: test] Error 1
make[1]: *** [Makefile:165: receiver/googlecloudspannerreceiver] Error 2
```

Another one https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10937096708/job/30362176527#step:8:232
```
=== Failed
make[1]: *** [Makefile:187: receiver/googlecloudspannerreceiver] Error 2
=== FAIL: internal/filter TestItemCardinalityFilter_Filter (0.05s)
make[1]: *** Waiting for unfinished jobs....
    logger.go:146: 2024-09-19T08:24:10.355Z	DEBUG	Added item to cache	{"seriesKey": "key1", "timestamp": "2021-10-13T20:30:00.000Z"}
    logger.go:146: 2024-09-19T08:24:10.408Z	DEBUG	Added item to cache	{"seriesKey": "key2", "timestamp": "2021-10-13T20:30:00.000Z"}
    logger.go:146: 2024-09-19T08:24:10.408Z	DEBUG	Skip item	{"seriesKey": "key3", "timestamp": "2021-10-13T20:30:00.000Z"}
    logger.go:146: 2024-09-19T08:24:10.408Z	DEBUG	Added item to cache	{"seriesKey": "key1", "timestamp": "2021-10-13T20:30:05.000Z"}
    logger.go:146: 2024-09-19T08:24:10.408Z	DEBUG	Added item to cache	{"seriesKey": "key5", "timestamp": "2021-10-13T20:30:05.000Z"}
    logger.go:146: 2024-09-19T08:24:10.408Z	DEBUG	Added item to cache	{"seriesKey": "key4", "timestamp": "2021-10-13T20:30:10.000Z"}
    logger.go:146: 2024-09-19T08:24:10.408Z	DEBUG	Added item to cache	{"seriesKey": "key6", "timestamp": "2021-10-13T20:30:10.000Z"}
    testhelpers_test.go:46: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter/testhelpers_test.go:46
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go:136
        	Error:      	Not equal: 
        	            	expected: 7
        	            	actual  : 8
        	Test:       	TestItemCardinalityFilter_Filter
```

--- a/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go
+++ b/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go
@@ -179,6 +179,9 @@ func TestItemCardinalityFilter_Filter(t *testing.T) {
 }
 
 func TestItemCardinalityFilter_FilterItems(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test on Windows due to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32397")
+	}
 	items := initialItemsWithSameTimestamp(t)
 	logger := zaptest.NewLogger(t)
 	filter, err := NewItemCardinalityFilter(metricName, totalLimit, limitByTimestamp, itemActivityPeriod, logger)

--- a/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go
+++ b/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go
@@ -6,6 +6,7 @@ package filter
 import (
 	"testing"
 	"time"
+	"runtime"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
@@ -122,6 +123,9 @@ func TestItemCardinalityFilter_Shutdown(t *testing.T) {
 }
 
 func TestItemCardinalityFilter_Filter(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test on Windows due to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32397")
+	}
 	items := initialItems(t)
 	logger := zaptest.NewLogger(t)
 	filter, err := NewItemCardinalityFilter(metricName, totalLimit, limitByTimestamp, itemActivityPeriod, logger)

--- a/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go
+++ b/receiver/googlecloudspannerreceiver/internal/filter/itemcardinality_test.go
@@ -4,9 +4,9 @@
 package filter
 
 import (
+	"runtime"
 	"testing"
 	"time"
-	"runtime"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"

